Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -2013,10 +2014,10 @@ void FitPeaks::generateCalculatedPeaksWS() {
   // create a wokspace with same number of input matrix workspace
   m_fittedPeakWS = create<Workspace2D>(*m_inputMatrixWS);
   for (size_t iws = 0; iws < m_fittedPeakWS->getNumberHistograms(); ++iws) {
-    auto out_vecx = m_fittedPeakWS->histogram(iws).x();
-    auto in_vecx = m_inputMatrixWS->histogram(iws).x();
+    auto &out_vecx = m_fittedPeakWS->mutableX(iws);
+    const auto &in_vecx = m_inputMatrixWS->x(iws);
     for (size_t j = 0; j < out_vecx.size(); ++j) {
-      m_fittedPeakWS->dataX(iws)[j] = in_vecx[j];
+      out_vecx[j] = in_vecx[j];
```

Comment:
Why is this not just assigning the complete X vector? It looks like it is the same size?

And if that is true: `create<>` as used above should actually keep the X of the parrent (since the size matches and no histogram is given), so in that case it should be possible to remove this whole assignment loop.

Reply:
That is actually true, the entire loop can be removed thanks to `create<>`. I somehow misread the inner loop and thought that only part of `in_vecx` was written to `out_vecx`.